src/commands/scan.ts

```diff
@@ -0,0 +1,139 @@
+import {Command, flags} from '@oclif/command';
+import {Scanner, ScanResult} from '../scanner';
+import {ResourceGraphTarget, DummyTarget, Target, RuleType} from '../rules';
+import cli from 'cli-ux';
+import chalk = require('chalk');
+
+interface LogOptions {
+  color?: string;
+  indent?: number;
+  bold?: boolean;
+}
+
+export default class Scan extends Command {
+  private isDebugMode = false;
+  private isVerbose = false;
+
+  static description =
+    'Scans Azure resources for potential configuration issues';
+
+  static examples = [
+    `$ aza scan --scope <SCOPE>
+    [rule-name]
+        [✓ | ❌][rule-description]     
+        Resources:
+                [resource-ids]
+
+    [total-passing]
+    [total-failing]
+    [total-rules-scanned]   
+`,
+  ];
+
+  static flags = {
+    help: flags.help({char: 'h'}),
+    scope: flags.string({
+      char: 's',
+      description: 'Azure subscription id to scan',
+    }),
+    dummy: flags.boolean({
+      char: 'd',
+      description: 'runs dummy rules to mock multi rule system',
+    }),
+    file: flags.string({
+      char: 'f',
+      description: 'JSON rules file path',
+    }),
+    verbose: flags.boolean({
+      char: 'v',
+      description: 'prints all results',
+    }),
+    debug: flags.boolean({
+      description: 'prints debugging logs',
+    }),
+  };
+
+  public log(message: string, options?: LogOptions) {
+    let formattedMessage = message;
+    if (options?.color) {
+      formattedMessage = chalk.keyword(options.color)(formattedMessage);
+    }
+    if (options?.indent) {
+      formattedMessage = ' '.repeat(options.indent) + formattedMessage;
+    }
+    if (options?.bold) {
+      formattedMessage = chalk.bold(formattedMessage);
+    }
+    super.log(formattedMessage);
+  }
+
+  private printResult(result: ScanResult) {
+    this.log(result.ruleName, {bold: true, indent: 4});
+    if (result.total) {
+      this.log(`❌ ${result.description}`, {color: 'grey', indent: 6});
+      this.log(`Resources (${result.total}):`, {indent: 6});
+      for (const id of result.resourceIds) {
+        this.log(id, {indent: 8});
+      }
+    } else {
+      this.log(`${chalk.green('✓')} ${chalk.grey(result.description)}`, {
+        indent: 6,
+      });
+    }
+    this.log('');
+  }
+
+  private printSummary(results: ScanResult[]) {
+    const total = results.length;
+    const totalPassed = results.filter(r => !r.total).length;
+    const totalFailed = total - totalPassed;
+    this.log(`${totalPassed} passing`, {
+      color: 'green',
+    });
+    if (this.isVerbose) {
+      this.log(`${totalFailed} failing`, {
+        color: 'red',
+      });
+    }
+    this.log(`${total} scanned`);
+  }
+
+  private print(results: ScanResult[]) {
+    for (const r of results) {
```

Comment:
IMO we want to print the failures always. It's not particularly useful if we say "something is broken" and don't tell the user what

Reply:
I agree. Let's get rid of that.